src/Evolution/DiscontinuousGalerkin/LiftFromBoundary.hpp

```diff
@@ -0,0 +1,179 @@
+    const gsl::span<const double>& boundary_corrections,
+    const DataVector& boundary_lifting_term,
+    const Scalar<DataVector>& magnitude_of_face_normal,
+    const Scalar<DataVector>& face_det_jacobian) noexcept;
+
+template <size_t Dim>
+void lift_boundary_terms_gauss_points_impl(
+    gsl::not_null<double*> volume_dt_vars, size_t num_independent_components,
+    const Mesh<Dim>& volume_mesh, size_t dimension,
+    const gsl::span<const double>& volume_det_inv_jacobian,
+    size_t num_boundary_pts,
+    const gsl::span<const double>& upper_boundary_corrections,
+    const DataVector& upper_boundary_lifting_term,
+    const Scalar<DataVector>& upper_magnitude_of_face_normal,
+    const Scalar<DataVector>& upper_face_det_jacobian,
+    const gsl::span<const double>& lower_boundary_corrections,
+    const DataVector& lower_boundary_lifting_term,
+    const Scalar<DataVector>& lower_magnitude_of_face_normal,
+    const Scalar<DataVector>& lower_face_det_jacobian) noexcept;
+}  // namespace detail
+
+/*!
+ * \brief Lift the boundary corrections to the volume time derivatives in the
+ * specified direction.
+ *
+ * The `boundary_corrections` are lifted to the `dt_vars` from the face in the
+ * direction `direction`. The magnitude of the normal vector in the \f$\xi\f$
+ * direction, for example, is given by
+ *
+ * \f{align*}{
+ *  \sqrt{\frac{\partial \xi}{\partial x^i} \gamma^{ij}
+ *  \frac{\partial \xi}{\partial x^j}}
+ * \f}
+ *
+ * The general lifting term (for the upper \f$\xi\f$-direction) is:
+ *
+ * \f{align*}{
+ * \partial_t u_{\breve{\imath}\breve{\jmath}\breve{k}}=\cdots
+ * -\frac{\ell_{\breve{\imath}}\left(\xi=1\right)}
+ *   {w_{\breve{\imath}}J_{\breve{\imath}\breve{\jmath}\breve{k}}}
+ *   \left[J\sqrt{
+ *   \frac{\partial\xi}{\partial x^i} \gamma^{ij}
+ *   \frac{\partial\xi}{\partial x^j}}
+ *   \left(G + D\right)
+ *   \right]_{\breve{\jmath}\breve{k}}\left(\xi=1\right),
+ * \f}
+ *
+ * where \f$\breve{\imath}\f$, \f$\breve{\jmath}\f$, and \f$\breve{k}\f$ are
+ * indices in the logical \f$\xi\f$, \f$\eta\f$, and \f$\zeta\f$ directions.
+ * The \f$G+D\f$ term corresponds to the `boundary_corrections`  function
+ * argument, and the function Spectral::boundary_lifting_term() is used to
+ * compute and cache the terms from the lifting term
+ * \f$\ell_{\breve{\imath}}(\xi=1)/w_{\breve{\imath}}\f$.
+ *
+ * \note that normal vectors are pointing out of the element and therefore the
+ * boundary term has an overall negative sign.
+ */
+template <size_t Dim, typename DtTagsList, typename BoundaryCorrectionTagsList>
+void lift_boundary_terms_gauss_points(
+    const gsl::not_null<Variables<DtTagsList>*> dt_vars,
+    const Scalar<DataVector>& volume_det_inv_jacobian,
+    const Mesh<Dim>& volume_mesh, const Direction<Dim>& direction,
+    const Variables<BoundaryCorrectionTagsList>& boundary_corrections,
+    const Scalar<DataVector>& magnitude_of_face_normal,
+    const Scalar<DataVector>& face_det_jacobian) noexcept {
```

Comment:
should probably be consistently "boundary," but the historical thing is that a lot of the domain code calls things "faces of the element" but in the equations you have a "boundary integral" not a "face integral" (or at least I've never encountered surface integrals dubbed "face integrals"). I'm hoping that I somewhat consistently use "face" for geometric-type quantities because of the "face" terminology in the domain, and "boundary" for variables/quantities associated with the PDEs themselves, rather than the details of the discretization and the geometric factors that arise from the discretization. My best guess as to why things are called "face" in the domain is to try and distinguish between the entire boundary of the element and just one part of it (a face), but I have no idea.

I'm happy to change terms to `boundary_det_jacobian` etc. though propagating that through everywhere I would like to not really do all at once... In short: I agree it's inconsistent but I'm not super enthusiastic about the tedious work to make it consistent, partly because I worry I'll have to deal with a lot of "Oh but I like the current thing" and that's exhausting...

Reply:
Thanks for the explanation - I think that's a reasonable way of differentiating. I agree it's not great to have the two words, but as long as I have a system to think about when to use one vs the other, I can handle it :P.

I think a switch to a single consistent name, if we decide to do it, is perhaps best saved for the future once all your PRs are in?